File: DiaryNutrition/app/src/main/java/ediger/diarynutrition/diary/water/WaterViewModel.java
package ediger.diarynutrition.diary.water;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Transformations;

import java.util.Calendar;
import java.util.List;

import ediger.diarynutrition.data.DiaryRepository;
import ediger.diarynutrition.data.source.entities.Water;
import ediger.diarynutrition.AppContext;

public class WaterViewModel extends AndroidViewModel {

    private final DiaryRepository mRepository;

    private final MutableLiveData<Calendar> mDate = new MutableLiveData<>();

    private final LiveData<List<Water>> mWater;

    public WaterViewModel(@NonNull Application application) {
        super(application);
        mRepository = ((AppContext) application).getRepository();
        mDate.setValue(Calendar.getInstance());

        mWater = Transformations.switchMap(mDate, mRepository::getWaterList);
    }

    public void setDate(Calendar day) {
        if (day != null) {
            mDate.setValue(day);
        }
    }

    public LiveData<List<Water>> getWater() {
        return mWater;
    }

    public void deleteWater(int id) {
        mRepository.deleteWater(id);
    }
}

File: DiaryNutrition/app/src/main/java/ediger/diarynutrition/diary/water/WaterViewModel.kt
package ediger.diarynutrition.diary.water

import android.app.Application
import androidx.lifecycle.*
import ediger.diarynutrition.AppContext
import ediger.diarynutrition.data.source.entities.Water
import kotlinx.coroutines.launch
import java.util.*

class WaterViewModel(application: Application) : AndroidViewModel(application) {

    private val waterRepository = (application as AppContext).waterRepository

    private val date = MutableLiveData<Calendar>()

    val water: LiveData<List<Water>>

    var addTime: Calendar = Calendar.getInstance()

    init {
        date.value = Calendar.getInstance()
        water = date.switchMap { day: Calendar -> waterRepository.getWaterList(day) }
    }

    fun setDate(day: Calendar) {
        date.value = day
    }

    fun addWater(amount: Int) = viewModelScope.launch {
        waterRepository.addWater(
                Water(amount, addTime.timeInMillis)
        )
    }

    fun deleteWater(id: Int) = viewModelScope.launch { waterRepository.deleteWater(id) }
}